Code/Tools/ProjectManager/Source/CreateAGemScreen.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#if !defined(Q_MOC_RUN)
+#include <ScreenWidget.h>
+#include <FormFolderBrowseEditWidget.h>
+#include <FormLineEditWidget.h>
+
+#include <QApplication>
+#include <QStyleOptionTab>
+#include <QStylePainter>
+#include <QTabBar>
+#include <QTabWidget>
+#include <QDialogButtonBox>
+#include <QPushButton>
+#include <QScrollArea>
+
+
+#endif
+
+class TabBar : public QTabBar
+{
+public:
+    QSize tabSizeHint(int index) const
+    {
+        QSize s = QTabBar::tabSizeHint(index);
+        s.transpose();
+        return s;
+    }
+
+protected:
+    void paintEvent(QPaintEvent* /*event*/)
```

Comment:
I still feel like there would be a better way to do this and have it automatically use the styles in the qss in the process, I'll investigate.

Reply:
+1

Reply:
Will stick to using paintEvent for vertical tab widget.